crates/bevy_reflect/src/serde/de.rs

```diff
@@ -12,6 +15,21 @@ pub trait DeserializeValue {
     ) -> Result<Box<dyn Reflect>, erased_serde::Error>;
 }
 
+/// A general purpose deserializer for reflected types.
+///
+/// For non-value types, this will return the dynamic equivalent. For example, a
+/// deserialized struct will return a [`DynamicStruct`] and a `Vec` will return a
+/// [`DynamicList`].
+///
+/// The serialized data must take the form of a map containing the following entries:
+/// 1. `type`: The _full_ [type name]
+/// 2. `value`: The serialized value of the reflected type
+///
+/// > Note: The ordering is important here. `type` _must_ come before `value`.
+///
+/// [`DynamicStruct`]: crate::DynamicStruct
+/// [`DynamicList`]: crate::DynamicList
+/// [type name]: std::any::type_name
 pub struct ReflectDeserializer<'a> {
```

Comment:
No strong feelings on the rename; I can see the arguments on both sides.

Reply:
I'm in favor of renaming to `Untyped`; it's clearer and indicates that there is at least one other option.

Can `TypedReflectDeserializer` deserialize to concrete types, since it has type info up front? If so, I think it would become the default deserializer, since that is what I and I think anyone else would expect a deserializer to do.

Reply:
Went ahead with the rename!